File: src/main/java/it/forgottenworld/fwparties/command/ChatCommand.java
package it.forgottenworld.fwparties.command;

import it.forgottenworld.fwparties.FWParties;
import it.forgottenworld.fwparties.controller.ChatController;
import it.forgottenworld.fwparties.controller.PartyController;
import it.forgottenworld.fwparties.util.TextUtility;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

import java.util.Arrays;

public class ChatCommand implements CommandExecutor {

    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (sender instanceof Player) {
            Player player = (Player) sender;
            PartyController partyController = FWParties.getInstance().getPartyController();
            ChatController chatController = FWParties.getInstance().getChatController();
            if (partyController.isPlayerInParty(player.getUniqueId())) {
                if (args.length > 0) {
                    String message = String.join(" ", Arrays.asList(Arrays.copyOfRange(args, 0, args.length)));
                    chatController.sendMessageToPartyMembers(partyController.getPlayerParty(player.getUniqueId()).getLeader(), "&2[PARTY] &a" + player.getName() + ": " + message);
                } else {
                    if (chatController.isPlayerChatting(player.getUniqueId())) {
                        player.sendMessage(TextUtility.parseColors("&eParty chat disabilitata!"));
                        chatController.removeChattingPlayer(player.getUniqueId());
                    } else {
                        player.sendMessage(TextUtility.parseColors("&eParty chat abilitata!"));
                        chatController.addChattingPlayer(player.getUniqueId());
                    }
                }
            } else {
                player.sendMessage(TextUtility.parseColors(FWParties.getInstance().getPluginConfig().getConfig().getString("not_on_party")));
            }
        } else {
            sender.sendMessage("Only players can run that command");
        }
        return true;
    }
}

File: src/main/kotlin/it/filippocavallari/command/ChatCommand.kt
package it.filippocavallari.command

import it.filippocavallari.Parties
import it.filippocavallari.utils.TextUtils
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class ChatCommand : SubCommand() {
    override fun getPermission(): String = "parties.chat"

    override fun onCommand(sender: CommandSender?, cmd: Command?, label: String?, args: Array<out String>) {
        if (sender is Player) {
            val partyManager = Parties.INSTANCE.partyManager
            val chatManager = Parties.INSTANCE.chatManager
            if (partyManager.isPlayerInParty(sender.uniqueId)) {
                if (args.isNotEmpty()) {
                    val message = java.lang.String.join(" ", listOf(*args.copyOfRange(0, args.size)))
                    partyManager.getPlayerParty(sender.uniqueId)?.let {
                        chatManager.sendMessageToPartyMembers(
                            it.leader,
                            "&2[PARTY] &a" + sender.name + ": " + message
                        )
                    }
                } else {
                    if (chatManager.isPlayerChatting(sender.uniqueId)) {
                        sender.sendMessage(TextUtils.parseColors("&eParty chat disabled!"))
                        chatManager.removeChattingPlayer(sender.uniqueId)
                    } else {
                        sender.sendMessage(TextUtils.parseColors("&eParty chat enabled!"))
                        chatManager.addChattingPlayer(sender.uniqueId)
                    }
                }
            } else {
                sender.sendMessage(TextUtils.parseColors(Parties.INSTANCE.config.config.getString("not_on_party")))
            }
        } else {
            sender?.sendMessage("Only players can run that command")
        }
    }
}